pkg/types/validation/installconfig.go

```diff
@@ -122,6 +123,13 @@ func ValidateInstallConfig(c *types.InstallConfig, usingAgentMethod bool) field.
 	allErrs = append(allErrs, validatePlatform(&c.Platform, usingAgentMethod, field.NewPath("platform"), c.Networking, c)...)
 	if c.ControlPlane != nil {
 		allErrs = append(allErrs, validateControlPlane(&c.Platform, c.ControlPlane, field.NewPath("controlPlane"))...)
+		if len(c.ControlPlane.Fencing.Credentials) > 0 {
+			if c.EnabledFeatureGates().Enabled(features.FeatureGateDualReplica) {
+				allErrs = append(allErrs, validateFencingCredentials(c)...)
```

Comment:
I would move this in to the validateControlPlane function:

https://github.com/openshift/installer/blob/513176f475eb1c8e840888c86e53a72a884bf6be/pkg/types/validation/installconfig.go#L747